tests/unit/ReferenceListTest.php

```diff
@@ -372,14 +372,4 @@ public function testGivenNonEmptyList_isNotEmpty() {
 		$this->assertFalse( $references->isEmpty() );
 	}
 
-	public function testGivenNonEmptyListWithForwardedIterator_isNotEmpty() {
-		$references = new ReferenceList();
-		$references->addNewReference( new PropertyNoValueSnak( 1 ) );
-		$references->next();
```

Comment:
See https://github.com/wmde/WikibaseDataModel/pull/609/files#r52314875

In release notes we say "ReferenceList no longer derives from SplObjectStorage, though still implements Countable and Traversable" which implies it does no longer implement the other interfaces of SplObjectStorage. Also please avoid comments like "Oh, wow" which are kind of disrespectful. We are not entirely stupid...